src/main/java/com/inscopemetrics/mad/statistics/StatisticFactory.java

```diff
@@ -60,80 +59,87 @@ public Statistic getStatistic(final String name) {
         final Statistic existingStatistic =  map.get(key);
         if (existingStatistic != null) {
             if (!existingStatistic.equals(statistic)) {
-                LOGGER.error()
-                    .setMessage("Statistic already registered")
-                    .addData("key", key)
-                    .addData("existing", existingStatistic)
-                    .addData("new", statistic)
-                    .log();
+                LOGGER.error(String.format(
+                        "Statistic already registered; key=%s, existing=%s, new=%s",
+                        key,
+                        existingStatistic,
+                        statistic));
             }
             return;
         }
         map.put(key, statistic);
     }
 
-    private static final ImmutableMap<String, Statistic> STATISTICS_BY_NAME_AND_ALIAS;
-    private static final ImmutableSet<Statistic> ALL_STATISTICS;
-    private static final InterfaceDatabase INTERFACE_DATABASE = ReflectionsDatabase.newInstance();
+    private static final Pattern PERCENTILE_STATISTIC_PATTERN = Pattern.compile("^[t]?p(?<percentile>[0-9]+(?:(\\.|p)[0-9]+)?)$");
+    private static final ConcurrentMap<String, Statistic> STATISTICS_BY_NAME_AND_ALIAS;
     private static final Logger LOGGER = LoggerFactory.getLogger(StatisticFactory.class);
 
     static {
         // NOTE: Do not put log messages in static blocks since they can lock the logger thread!
-        final Map<String, Statistic> statisticByNameAndAlias = Maps.newHashMap();
-        final Set<Statistic> allStatistics = Sets.newHashSet();
-        final Set<Class<? extends Statistic>> statisticClasses = INTERFACE_DATABASE.findClassesWithInterface(Statistic.class);
-        for (final Class<? extends Statistic> statisticClass : statisticClasses) {
-            if (!statisticClass.isInterface() && !Modifier.isAbstract(statisticClass.getModifiers())) {
-                try {
-                    final Constructor<? extends Statistic> constructor = statisticClass.getDeclaredConstructor();
-                    if (!constructor.isAccessible()) {
-                        constructor.setAccessible(true);
-                    }
-                    final Statistic statistic = constructor.newInstance();
-                    allStatistics.add(statistic);
-                    checkedPut(statisticByNameAndAlias, statistic, statistic.getName());
-                    for (final String alias : statistic.getAliases()) {
-                        checkedPut(statisticByNameAndAlias, statistic, alias);
+        final ConcurrentMap<String, Statistic> statisticByNameAndAlias = Maps.newConcurrentMap();
+        try {
+            final ImmutableSet<ClassPath.ClassInfo> statisticClasses = ClassPath.from(StatisticFactory.class.getClassLoader())
+                    .getTopLevelClasses("com.inscopemetrics.mad.statistics");
```

Comment:
I'm not a fan of this at all.  This will be an issue if we want to have extension jars.